server/room.ts

```diff
@@ -29,212 +36,124 @@ export class Room {
-        io.of(roomId).emit('roster', this.roster);
-      });
-      socket.on('CMD:startVBrowser', async () => {
-        if (this.vBrowser) {
-          // Maybe terminate the existing instance and spawn a new one
-          return;
-        }
-        this.cmdHost(socket, 'vbrowser://');
-        this.vBrowser = {};
-        const assignment = await assignVM();
-        if (!assignment) {
-          this.cmdHost(socket, '');
-          this.vBrowser = undefined;
-          return;
-        }
-        const { pass, host, id } = assignment;
-        this.vBrowser.assignTime = Number(new Date());
-        this.vBrowser.pass = pass;
-        this.vBrowser.host = host;
-        this.vBrowser.id = id;
-        this.roster.forEach((user, i) => {
-          if (user.id === socket.id) {
-            this.roster[i].isController = true;
-          } else {
-            this.roster[i].isController = false;
-          }
-        });
-        this.cmdHost(
-          undefined,
-          'vbrowser://' + this.vBrowser.pass + '@' + this.vBrowser.host
-        );
-        io.of(roomId).emit('roster', this.roster);
-      });
-      socket.on('CMD:stopVBrowser', () => this.resetRoomVM());
-      socket.on('CMD:changeController', (data: string) => {
-        this.roster.forEach((user, i) => {
-          if (user.id === data) {
-            this.roster[i].isController = true;
-          } else {
-            this.roster[i].isController = false;
-          }
-        });
-        io.of(roomId).emit('roster', this.roster);
-      });
-      socket.on('CMD:askHost', () => {
-        socket.emit('REC:host', this.getHostState());
-      });
-      socket.on('signal', (data: { to: string; msg: string }) => {
-        io.of(roomId)
-          .to(data.to)
-          .emit('signal', { from: socket.id, msg: data.msg });
-      });
-      socket.on(
-        'signalSS',
-        (data: { to: string; sharer: boolean; msg: string }) => {
-          io.of(roomId).to(data.to).emit('signalSS', {
-            from: socket.id,
-            sharer: data.sharer,
-            msg: data.msg,
-          });
-        }
-      );
+  nextVideo() {
+    this.video = undefined;
 
-      socket.on('disconnect', () => {
-        let index = this.roster.findIndex((user) => user.id === socket.id);
-        const removed = this.roster.splice(index, 1)[0];
-        io.of(roomId).emit('roster', this.roster);
-        if (removed.isScreenShare) {
-          // Reset the room state since we lost the screen sharer
-          this.cmdHost(socket, '');
-        }
-        delete this.tsMap[socket.id];
-        // delete nameMap[socket.id];
-      });
+    if (this.videoPlaylist.length > 0) {
+      const nextVideo = this.videoPlaylist.splice(0, 1);
+      this.video = nextVideo[0].url;
+      for (const connection of this.connections) {
+        connection.socket.emit('playlistUpdate', this.videoPlaylist);
+        this.cmdHost(connection.socket, nextVideo[0], { skipMessage: true });
```

Comment:
should this be in the for loop? Generally cmdhost should only be done once per video change